tests/end-to-end/Smoke.Tests/BasicConversions.cs

```diff
@@ -21,10 +21,18 @@ public class BasicSmokeTests : IClassFixture<SolutionPathFixture>, IClassFixture
         private string GetFSharpProjectPath(string projectName) => Path.Combine(TestDataPath, projectName, $"{projectName}.fsproj");
         private string GetCSharpProjectPath(string projectName) => Path.Combine(TestDataPath, projectName, $"{projectName}.csproj");
         private string GetVisualBasicProjectPath(string projectName) => Path.Combine(TestDataPath, projectName, $"{projectName}.vbproj");
+        private string GetXamarinAndroidProjectPath(string projectName) => Path.Combine(TestDataPath, projectName, $"{projectName}.csproj");
+        private string GetXamariniOSProjectPath(string projectName) => Path.Combine(TestDataPath, projectName, $"{projectName}.csproj");
+        //set to true if running Xamarin Project Tests
+        private bool isXamarinTest => true;
```

Comment:
Instead of doing this during test construction, what if the xamarin tests themselves just call a method that re-registers the msbuild to be resolved for the test? That would get rid of the global flag.

Reply:
Refactored tests! there's a separate project now for the MAUI Conversion test. Trying to re-register msbuild in the same project was kind of messy, based on race conditions, one or more tests would fail with this error 

`System.InvalidOperationException : Microsoft.Build.Locator.MSBuildLocator.RegisterInstance was called, but MSBuild assemblies were already loaded.
    Ensure that RegisterInstance is called before any method that directly references types in the Microsoft.Build namespace has been called.
    This dependency arises from when a method is just-in-time compiled, so if it breaks even if the reference to a Microsoft.Build type has not been executed.
    For more details, see aka.ms/RegisterMSBuildLocator
    `
    
    So least destructive path forward was to make a separate project for these tests 😄